File: sk000801/LEVEL1/기사단원의 무기.java
class Solution {
    
    public int countDivisors(int n) {
            int cnt = 0;
             for (int i = 1; i <= Math.sqrt(n); i++) {
                if (n % i == 0) {
                    if (n / i == i) {
                        cnt++;
                    }
                    else {cnt += 2;}
                }
            }
            return cnt;
    }
         
    
    public int solution(int number, int limit, int power) {
        int answer = 0;
        for(int i = 1; i <= number; i++) {
           int divisor = countDivisors(i);
           if (divisor > limit) {
               answer += power;
           } else {
               answer += divisor;
           }
        }
        return answer;
    }
}

File: kyw0716/level 1/기사단원의 무기.js
function YaksooCount(number) {
  let count = 0;
  let isOdd =
    Math.sqrt(number) - Math.floor(Math.sqrt(number)) === 0 && number !== 1;

  for (let i = 1; i <= Math.sqrt(number); i++) {
    if (number % i === 0) count += 1;
  }

  return isOdd ? 2 * count - 1 : 2 * count;
}

function solution(number, limit, power) {
  var answer = 1;

  for (let i = 2; i <= number; i++) {
    const yaksoo = YaksooCount(i);
    if (yaksoo > limit) {
      answer += power;
      continue;
    }
    answer += yaksoo;
  }

  return answer;
}

// https://school.programmers.co.kr/learn/courses/30/lessons/136798
